### ISSUE TITLE
build-deb.sh fails to run

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#3771]
Raised in https://github.com/silx-kit/silx/issues/3443#issuecomment-1452052407 by @tobiasvos:

I get the following error when trying to build a Debian package (and even with just `./build-deb.sh --help`):

Fatal Python error: init_sys_streams: can't initialize sys standard streams
Python runtime state: core initialized
```
Traceback (most recent call last):
  File "silx/src/silx/io/__init__.py", line 36, in <module>
  File "silx/src/silx/io/utils.py", line 34, in <module>
  File "/usr/lib/python3.9/logging/__init__.py", line 26, in <module>
  File "/usr/lib/python3.9/traceback.py", line 5, in <module>
  File "/usr/lib/python3.9/linecache.py", line 11, in <module>
  File "/usr/lib/python3.9/tokenize.py", line 27, in <module>
ImportError: cannot import name 'open' from 'builtins' (unknown location)
```
The issue is that the silx/src/io subpackage shadows the standard library's io module. This issue can be circumvented by renaming src/silx/io to e.g. src/silx/_io, but then the build process still fails because silx.io.xxx is expected in many places.

Any clue how to get around this?

**Comment by t20100** [comment]
We never heard of such issue..

I just added debian11 to our CI and it seems to work fine (I haven't tested the generated package, but there are tests in the build process).
You can download the packages for v1.1.2 from: https://gitlab.esrf.fr/silx/bob/silx/-/jobs/579821/artifacts/browse/artifacts/bullseye/

If even `./build-deb.sh --help` fails, then I expect the issue to come from:
https://github.com/silx-kit/silx/blob/3a0a265fbd50ac66d94d0481c455fa1b6d11c084/build-deb.sh#L67-L70
Can you try to run `python3 -c"import _version; print(_version.version)"` from `silx/src/silx` to confirm it?

Indeed if this needs to `import io` it will cause trouble!
Do you start this script in any special way (from an IDE GUI?) that might change the way the Python interpreter is initialized?

**Comment by tobiasvos** [comment]
Thank you for your help. I have installed the generated package for Debian 11 and it seems to work fine.

> If even `./build-deb.sh --help` fails, then I expect the issue to come from:
> 
> https://github.com/silx-kit/silx/blob/3a0a265fbd50ac66d94d0481c455fa1b6d11c084/build-deb.sh#L67-L70
> 
> Can you try to run `python3 -c"import _version; print(_version.version)"` from `silx/src/silx` to confirm it?

Good find! I can confirm that this is the culprit. Also, when running `./build-deb.sh --help`, I see the abovementioned error/traceback three times, so once for lines 68, 69 and 70.

In fact, I get the same error when running the Python interpreter without any arguments (`python3`) from `silx/src/silx`. Apparently, on my machine the io module is automatically loaded. When I do [this](https://stackoverflow.com/a/7644454), the list includes 'io'. I tested this on Debian 9, 10 and 11. Apparently it is built into the interpreter that I use, which on Debian 11 is the `/usr/bin/python3` provided by the [python3-minimal package](https://packages.debian.org/bullseye/python3-minimal). According to [here](https://docs.python.org/3/tutorial/modules.html#standard-modules), the set of such ("builtin" or "frozen") modules is a configuration option which also depends on the underlying platform.

In any case, my problem (not having access to a silx version newer than 0.14.0) is solved, so thanks again!
